data/src/main/kotlin/us/wedemy/eggeum/android/data/repository/LoginRepositoryProvider.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Designed and developed by Wedemy 2023.
+ *
+ * Licensed under the MIT.
+ * Please see full license: https://github.com/Wedemy/eggeum-android/blob/main/LICENSE
+ */
+
+package us.wedemy.eggeum.android.data.repository
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.adapter
+import io.ktor.client.HttpClient
+import io.ktor.client.request.post
+import io.ktor.client.statement.bodyAsText
+import javax.inject.Inject
+import javax.inject.Singleton
+import us.wedemy.eggeum.android.data.client.jsonBody
+import us.wedemy.eggeum.android.data.mapper.toDomain
+import us.wedemy.eggeum.android.data.model.login.LoginBodyResponse
+import us.wedemy.eggeum.android.data.model.login.SignUpBodyResponse
+import us.wedemy.eggeum.android.domain.model.login.LoginBody
+import us.wedemy.eggeum.android.domain.model.login.SignUpBody
+import us.wedemy.eggeum.android.domain.repository.LoginRepository
+
+@Singleton
+public class LoginRepositoryProvider @Inject constructor(
+  private val client: HttpClient,
+  moshi: Moshi,
+) : LoginRepository {
+  private val loginBodyAdapter = moshi.adapter<LoginBodyResponse>()
+  private val signUpBodyAdapter = moshi.adapter<SignUpBodyResponse>()
+
+  override suspend fun getLoginBody(idToken: String?): LoginBody? {
+    val responseText = client.post("app/sns-sign-in") {
+      jsonBody(true) {
+        "idToken" withString idToken
+      }
+    }.bodyAsText()
```

Comment:
이 부분 코드 스타일을 하나로 통일하는 게 좋을까요?
지훈님 의견이 궁금해요!

Reply:
네 통일하면 좋죠~

Reply:
```suggestion
    val responseText = 
      client
        .post("app/sns-sign-in") {
          jsonBody(true) {
            "idToken" withString idToken
          }
        }.bodyAsText()
```

그렇다면 저는 이렇게 하는 편이에요!

Reply:
오호 그릏군요 요렇게 개행 맞춰서 올려볼게요 .body() 로만 받았었는데 .bodyAsText() 가 있는지 이번에 알았네요

Reply:
그리고 추가적으로 질문이 있습니다 UseCase 에 @Singleton Annotation 을 붙힌건 이해를 했는데 RepositoryImpl 에서 @Singleton 을 붙히신 이유가 있을까요? 
data 모듈의 di 패키지의 RespositoryModule 에서 repository 의 주입을 필요로 하는 클래스들에게 @Singleton 으로 주입해주도록 이미 설정을 해놓은 상태인데 RespositoryImpl 에도 @Singleton 을 붙히는건 중복이지 않나? 라는 생각이 들어서요 UseCase 는 interface, impl 형태로 구현하지 않았기 때문에 di 패키지에 provide 함수를 정의해주지 않아서 붙혀주신건가 하고 전 이해를 했어서요!

Reply:
> 왜 일부러 Response 를 Text로 받아서 text를 adapter.fromJson(text) 을 통해 직렬화 해주는건지 알 수 있을까요?

Ktor Client에서 [Content negotiation and serialization](https://ktor.io/docs/serialization-client.html)이 Moshi 지원을 안 해요.

Reply:
> RepositoryImpl(xxxRepositoryProvider) 에서 @singleton 을 붙히신 이유가 있을까요?

도메인적 관점을 정확히 명시하기 위함이에요.

1. di container가 없더라도 이 친구는 싱글톤으로 제공되야 함을 나타내고,
2. 만약 di container로 제공된다 해도 해당 container가 이 친구를 싱글톤으로 취급해야 함을 나타냅니다.

Reply:
답변 감사합니당 ㅎㅎ

Reply:
이해는 모두 되셨을까용?

Reply:
넵넵!